### PR TITLE
AMPI: first pass of AMPI-only build optimizations

### PR DESCRIPTION
**Pull request by stwhite91**
AMPI doesn't use Charm-MPI interop or message priorities, so disable them.

### CODE REVIEW

**Comment by stwhite91** [comment]
Build failure is in megacon on netlrts-smp

File: build [hunk 1]
```diff
@@ -863,6 +863,8 @@ then
       # strip "-only" suffix from AMPI target
       PROGRAM="${PROGRAM%-only}"
     fi
+    OPTS="$OPTS -DCMK_NO_INTEROP=1 -DCMK_NO_MSG_PRIOS=1 "
+    CONFIG_OPTS="$CONFIG_OPTS --with-prio-type=char "
```
Comment (evan-charmworks) [inline build]: Maybe the `$OPTS` and `$CONFIG_OPTS` should be at the end so manually passed parameters can overwrite them.

Comment (stwhite91) [reply build]: Done

**Comment by matthiasdiener** [comment]
Should  be replicated in the cmake build

**Comment by stwhite91** [comment]
> Should be replicated in the cmake build

Done

**Pull request merged**